index.js

```diff
@@ -0,0 +1,84 @@
+'use strict'
+
+const path = require('path')
+const statSync = require('fs').statSync
+
+const send = require('send')
+
+const fp = require('fastify-plugin')
+
+const DEFAULT_500_PAGE = path.join(__dirname, 'static', '500.html')
+const DEFAULT_403_PAGE = path.join(__dirname, 'static', '403.html')
+const DEFAULT_404_PAGE = path.join(__dirname, 'static', '404.html')
+
+function fastifyStatic (fastify, opts, next) {
+  const error = checkOptions(opts)
+  if (error instanceof Error) return next(error)
+
+  const root = opts.root
+  const page500 = opts.page500Path || DEFAULT_500_PAGE
+  const page403 = opts.page403Path || DEFAULT_403_PAGE
+  const page404 = opts.page404Path || DEFAULT_404_PAGE
+
+  function overwriteStatusCode (res, statusCode) {
+    return function () { res.statusCode = statusCode }
+  }
+
+  function servePathWithStatusCodeWrapper (page, statusCode) {
+    return function servePage (req, res) {
+      send(req, page)
+        .on('stream', overwriteStatusCode(res, statusCode))
+        .pipe(res)
+    }
+  }
+  const serve404 = servePathWithStatusCodeWrapper(page404, 404)
+  const serve403 = servePathWithStatusCodeWrapper(page403, 403)
+  const serve500 = servePathWithStatusCodeWrapper(page500, 500)
```

Comment:
I am a bit lost on those. We should have these pages in fastify itself, rather than in here. How do we return a 404 page in fastify? A user should not configure this separately from the framework itself.

Reply:
Me too but consider that this is the 404 page only in the prefixed routes.

Reply:
IMHO, we should land this as it is, and then fix fastify - as these are no place here.